kernel/thor/generic/address-space.cpp

```
@@ -424,14 +424,12 @@ VirtualSpace::map(smarter::borrowed_ptr<MemorySlice> slice,
 		auto spaceLock = frg::guard(&_snapshotMutex);
 
 		if(flags & kMapFixed) {
-			assert(address);
 			assert((address % kPageSize) == 0);
 
 			actualAddress = _allocateAt(address, length);
 		}else{
 			actualAddress = _allocate(length, flags);
 		}
-		assert(actualAddress);
```

This should propagate the error back to user-space